src/middlewared/middlewared/plugins/virt/global.py

```diff
@@ -271,15 +303,112 @@ async def _setup_impl(self):
+        to deletion of an incus storage pool.
+        """
+        resp = await incus_call(f'1.0/storage-pools/{pool_name}', 'get')
+        if resp['type'] == 'error':
+            if resp['error_code'] == 404:
+                # storage doesn't exist. Nothing to do.
+                return []
+
+            raise CallError(resp['error'])
+
+        out = []
+
+        for dependent in resp['metadata']['used_by']:
+            if dependent.startswith(('/1.0/images/')):
+                continue
+
+            path = dependent.split('/')
+            if 'storage-pools' in path:
+                # sample:
+                # /1.0/storage-pools/dozer/volumes/custom/foo
+                incus_type = path[4]
+            else:
+                incus_type = path[2]
+
+            out.append({'type': incus_type, 'name': path[-1]})
+
+        return out
+
+    @private
+    async def remove_storage_pool(self, pool_name):
+        resp = await incus_call(f'1.0/storage-pools/{pool_name}', 'get')
+        if resp['type'] == 'error':
+            if resp['error_code'] == 404:
+                # storage doesn't exist. Nothing to do.
+                return
+
+            raise CallError(resp['error'])
+
+        to_delete = []
+
+        for dependent in resp['metadata']['used_by']:
+            # Middleware internally manages the images and profiles for
+            # storage pools
+            if dependent.startswith('/1.0/images/'):
+                to_delete.append(dependent)
+
+        if remainder := (set(resp['metadata']['used_by']) - set(to_delete)):
+            raise CallError(
+                f'Storage volume currently used by the following incus resource {", ".join(remainder)}', errno.EBUSY
+            )
+
+        for entry in to_delete:
+            path = entry[1:]  # remove leading slash
+            resp = await incus_call(path, 'delete')
+            if resp['type'] == 'error' and resp['error_code'] != 404:
+                raise CallError(f"{resp['error_code']}: {resp['error']}")
+
+        # Finally remove the pool itself
+        # We get intermittent errors here from incus API (appears to be replaying last command)
+        # unless we have a sleep
+        await asyncio.sleep(1)
+
+        resp = await incus_call(f'1.0/storage-pools/{pool_name}', 'delete')
+        if resp['type'] == 'error':
+            raise CallError(resp['error'])
+
+    async def _setup_impl(self):
+        config = await self.config()
+        to_import = []
+
+        if not config['pool']:
+            if await self.middleware.call('service.started', 'incus'):
+                job = await self.middleware.call('virt.global.reset', False, config)
                 await job.wait(raise_error=True)
 
+            self.logger.debug('No pool set for virtualization, skipping.')
+            raise NoPoolConfigured()
+        else:
+            await self.middleware.call('service.start', 'incus')
```

Comment:
Need to make sure this service.start doesn't propagate the command to the standby